xblock/runtime.py

```diff
@@ -1267,17 +1268,42 @@ def __getattr__(self, name):
         return getattr(self._translations, name)
 
     STRFTIME_FORMATS = {
-        "SHORT_DATE_FORMAT": "%b %d, %Y",
-        "LONG_DATE_FORMAT": "%A, %B %d, %Y",
-        "TIME_FORMAT": "%I:%M:%S %p",
-        "DATE_TIME_FORMAT": "%b %d, %Y at %H:%M",
+        u"SHORT_DATE_FORMAT": u"%b %d, %Y",
+        u"LONG_DATE_FORMAT": u"%A, %B %d, %Y",
+        u"TIME_FORMAT": u"%I:%M:%S %p",
+        u"DATE_TIME_FORMAT": u"%b %d, %Y at %H:%M",
     }
 
     def strftime(self, dtime, format):      # pylint: disable=redefined-builtin
         """
         Locale-aware strftime, with format short-cuts.
         """
         format = self.STRFTIME_FORMATS.get(format + "_FORMAT", format)
-        if isinstance(format, unicode):
-            format = format.encode("utf8")
-        return dtime.strftime(format).decode("utf8")
+        if future.utils.PY2 and isinstance(format, str):
+            format = format.encode("utf-8")
+        timestring = dtime.strftime(format)
+        if future.utils.PY2:
+            timestring = timestring.decode('utf-8')
+        return timestring
+
+    @property
+    def ugettext(self):
+        """
+        Dispatch to the appropriate gettext method
+        """
+        # pylint: disable=no-member
+        if future.utils.PY3:
```

Comment:
Since we are starting fresh here, the conditionals should all be on PY2, not PY3, for that day in the far future when Python 4.x is out, and behaves like Python 3 (which we have been promised it will).